### PR TITLE
fix: enable deployment on rdev

### PR DESCRIPTION
**Pull request by roaga**
#### Reviewers
**Functional:** 
@ebezzi 

**Readability:** 
@Bento007 

---

## Changes
- modify config files to allow Explorer to be deployed on rdev.
- modify Docker-related files to take advantage of caching and trim extra files.

### CODE REVIEW

File: hosted/Dockerfile [hunk 1]
```diff
@@ -1,40 +1,46 @@
-FROM ubuntu:focal
+FROM ubuntu:focal as build
 
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
 ARG HAPPY_ENV
 ENV HAPPY_ENV=$HAPPY_ENV
+ENV STACK_NAME="roaga-somaexplorer"
```
Comment (ebezzi) [inline hosted/Dockerfile:7]: Did the infra team reach out to you about whether this variable is available at build time?

File: hosted/Dockerfile [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
 RUN apt-get update && \
     apt-get install -y gettext moreutils build-essential libxml2-dev python3-dev python3-pip zlib1g-dev python3-requests python3-aiohttp llvm jq npm git zip && \
-    rm -rf /var/lib/apt/lists/*
-
-RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1
-
-RUN python3 -m pip install psycopg2-binary
+    rm -rf /var/lib/apt/lists/* && \
+    apt-get clean && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 && \
```
Comment (ebezzi) [inline hosted/Dockerfile:16]: Wondering if this is still necessary. I believe `focal` already has python3.8 as default.

File: hosted/Dockerfile [hunk 3]
```diff
@@ -1,40 +1,46 @@
...
+    python3 -m pip install psycopg2-binary
```
Comment (ebezzi) [inline hosted/Dockerfile:17]: I would remove this. `psycopg2` is a driver for Postgres which isn't used by the explorer.

File: hosted/Dockerfile (hunk 1 above)
Comment (roaga) [reply hosted/Dockerfile:7]: It's not available directly. "You could add those to your task definition as environment variables, which could then be used by your docker-compose." But that's not any easier for a user of this system than just changing this file. It is a feature they are thinking about though and are looking for feedback on.

File: docker-compose.yml [hunk 4]
```diff
@@ -28,7 +28,7 @@ services:
       context: .
       dockerfile: hosted/Dockerfile
       cache_from:
-        - "${DOCKER_REPO}corpora-explorer:branch-main"
+        - "${DOCKER_REPO}corpora-explorer:latest"
```
Comment (ebezzi) [inline docker-compose.yml:31]: Any reason for this?

Comment (roaga) [reply docker-compose.yml:31]: The tag on the images is "latest," not "branch-main," so when it tried to pull from cache, it often errored that it couldn't find it. Switching to "latest" fixes that.

Comment (ebezzi) [reply docker-compose.yml:31]: Ah sorry, this is just for caching. 👍 then

**Review by ebezzi — APPROVED** [review]

**Pull request merged**